### PR TITLE
#156 - Added skip_types and optimize options to rpm_sync

### LINKED ISSUE #156: Adding skip_types & optimize options to rpm_sync.py
When calling a synchronization of rpm repos, allow the passing of "skip_types" and "optimize" to be passed. PULP RPM module allows for this already.

Creating issue for tracking and will be working on a PR.

---

### PR DESCRIPTION
**Pull request by chtaylo2**
#156 

Adding support for "skip_types" and "optimize" options to rpm_sync. Created tests inside "rpm_sync.yaml". Finally, refreshed the fixtures to reference the new tests.

I've run the full set of tests and all are passing, 100%.

Thanks,
Chris

### CODE REVIEW

File: plugins/modules/rpm_sync.py [hunk 1]
```diff
@@ -106,6 +126,18 @@ def main():
             mirror = module.params["sync_policy"] == "mirror_complete"
             parameters = {"mirror": mirror}
 
+        parameters = {"optimize": module.params["optimize"]}
```
Comment (mdellweg) [inline plugins/modules/rpm_sync.py]: you are overwriting what we already collected in parameters here. This cannot possibly be correct.

**Comment by mdellweg** [comment]
Please add `fixes #156` to the commit message

File: plugins/modules/rpm_sync.py (hunk 1 above)
Comment (chtaylo2) [reply plugins/modules/rpm_sync.py]: Good call out, I'll fix that. I also realized there's no test for the sync_policy key:pair. I'll proactively add that in.

Comment (mdellweg) [reply plugins/modules/rpm_sync.py]: I guess you can fold that into the same update calls, no?

Comment (chtaylo2) [reply plugins/modules/rpm_sync.py]: Yep, exactly. I'll do that

**Comment by mdellweg** [comment]
Can you please squash the commits?

Did you know that running `make format` should help you getting the formatting right? (OK, it's missing from `make help`...)

**Comment by chtaylo2** [comment]
Rebase pushed to squash previous commits

**Comment by mdellweg** [comment]
Looks good now. But we need some engineering on the commit message.
Can you do `git commit --amend` and change that to
```
Add skip_types and optimize options to rpm_sync

fixes #156
```

**Comment by chtaylo2** [comment]
> Please add fixes #156 to the commit message

Got it. I misread your request above to add "more" fix details to the commit message. Glad to see you want this limited and details to remain in the issue tracker.  Commit message changed.

**Review by mdellweg — APPROVED** [review]

**Comment by mdellweg** [comment]
> > Please add fixes #156 to the commit message
> 
> Got it. I misread your request above to add "more" fix details to the commit message. Glad to see you want this limited and details to remain in the issue tracker. Commit message changed.

Well, i don't want to push too hard. Also there is not much more one can say anyway. But I am a big fan of good commit messages (https://cbea.ms/git-commit/) because that's the only thing really persisting. And i hate how GH UI is trying hard to draw every attention away from commit messages.

**Pull request merged**